UI/src/app/_directives/requireRole.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Directive,
+  Input,
+  OnInit,
+  TemplateRef,
+  ViewContainerRef
+} from '@angular/core';
+import { AuthService } from '../_services/auth.service';
+
+@Directive({
+  selector: '[requireRole]'
+})
+export class RequireRoleDirective implements OnInit {
+  @Input() requireRole = '';
+
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private authService: AuthService
+  ) {}
+
+  ngOnInit() {
+    if(this.authService.requireRole(this.requireRole)) {
+      this.viewContainer.createEmbeddedView(this.templateRef);
+    } else {
+      this.viewContainer.clear();
```

Comment:
What about displaying "Your user does not have the rights to access [name]" here?
Otherwise we might get a lot of issues like "Why can I see that part and he can't" because users might not understand the roles at first?

Reply:
Yes, I think we need to add roles explanations.
@hrueger @JTF4 can you take a look at it? I don't know how can I implement this.

Reply:
https://github.com/angular/angular/blob/13.0.3/packages/core/src/linker/view_container_ref.ts#L106-L128 that should do the trick. Just vreate a new component for that and insert it after the `clear()`.